src/http.rs

```diff
@@ -262,6 +499,65 @@ impl<B> fmt::Debug for HttpClient<B> {
     }
 }
 
+/// Configuration for HTTP client providing an authentication mechanism. 
+#[configurable_component]
+#[configurable(metadata(docs::advanced))]
+#[derive(Clone, Debug)]
+#[serde(deny_unknown_fields)]
+pub struct HttpClientAuthorizationConfig {
+    /// Define how to authorize against an upstream.
+    #[configurable]
+    auth: HttpClientAuthorizationStrategy,
```

Comment:
Can we flatten this to avoid repetition in configs, from: 
```
    http_client_authorization_strategy:
      auth:
        strategy: "o_auth2"
        client_id: "your.clientid"
        client_secret: "verysecretstring"
        token_endpoint: "https://your.provider.com/oauth/token"
```
to
```
    http_client_authorization_strategy:
      strategy: "o_auth2"
      client_id: "your.clientid"
      client_secret: "verysecretstring"
      token_endpoint: "https://your.provider.com/oauth/token"
```

I think this makes sense from a UX perspective cc @vectordotdev/ux-team

Reply:
Honestly I was unable to figure out this - @pront  can you pls point me some example with flattened enum ?

Reply:
I took another look at this:
* We can replace `pub http_client_authorization_strategy: Option<HttpClientAuthorizationConfig>` with `pub authorization_config: Option<AuthorizationConfig>`
* Rename `auth` to `strategy`

Reply:
then we end with double strategy.strategy, is that ok ?
```yaml
sinks:
  my_sink_id:
    authorization_config:
      strategy:
        strategy: "o_auth2"
        client_id: "some.client.id"
        client_secret: "ab!@#"
        # token_endpoint: "https://profile.api.stage.context.cloud.sap/uaa/oauth/token"
        grace_period: 780
        token_endpoint: "https://127.0.0.1:8091/oauth/token"
```
maybe I will change this tag 
```yaml
/// Configuration of the authentication strategy for HTTP requests.
///
/// HTTP authentication should be used with HTTPS only, as the authentication credentials are passed as an
/// HTTP header without any additional encryption beyond what is provided by the transport itself.
#[configurable_component]
#[derive(Clone, Debug)]
#[serde(deny_unknown_fields, rename_all = "snake_case", tag = "strategy")]
#[configurable(metadata(docs::enum_tag_description = "The authentication strategy to use."))]
```
to 'protocol' ?

Reply:
then config will look like this 
```yaml
sinks:
  my_sink_id:
    authorization_config:
      strategy:
        strategy: "o_auth2"
        client_id: "some.client.id"
        client_secret: "ab!@#"
        grace_period: 780
        token_endpoint: "https://127.0.0.1:8091/oauth/token"
```
is this double strategy ok ? maybe I will change also tag from 
```yaml
/// Configuration of the authentication strategy for HTTP requests.
///
/// HTTP authentication should be used with HTTPS only, as the authentication credentials are passed as an
/// HTTP header without any additional encryption beyond what is provided by the transport itself.
#[configurable_component]
#[derive(Clone, Debug)]
#[serde(deny_unknown_fields, rename_all = "snake_case", tag = "strategy")]
#[configurable(metadata(docs::enum_tag_description = "The authentication strategy to use."))]
```
to 'protocol'

Reply:
together with fix for mtls extension I pushed renamed struct 
```
#[configurable(derived)]
pub authorization_config: Option<AuthorizationConfig>,
```